index.bs

```diff
@@ -701,62 +707,103 @@ When this method is invoked, the user agent MUST execute the following algorithm
+                    and [=map/values=] of type [=client extension output=]. <code>[=extensionOutputsMap=]</code>'s
+                    [=map/entries=] are created by running each extension's [=client extension processing=] algorithm to create
+                    the [=client extension outputs=], for each [=client extension=] in
+                    <code>{{AuthenticatorResponse/clientDataJSON}}.clientExtensions</code>.
+
+            1.  Let |constructCredentialAlg| be an algorithm that takes a [=environment settings object/global object=]
+                |global|, and whose steps are:
+
+                1.  Let |attestationObject| be a new {{ArrayBuffer}}, created using |global|'s [=%ArrayBuffer%=], containing the
+                    bytes of <code>|credentialCreationData|.[=attestationObjectResult=]</code>'s value.
+
+                1.  Let |id| be <code>|attestationObject|.authData.[=attestedCredentialData=].[=credentialId=]</code>.
+
+                1.  Let |pubKeyCred| be a new {{PublicKeyCredential}} object associated with |global| whose fields are:
+
+                    :   {{PublicKeyCredential/[[identifier]]}}
+                    ::  |id|
+
+                    :   {{PublicKeyCredential/response}}
+                    ::  A new {{AuthenticatorAttestationResponse}} object associated with |global| whose fields are:
+
+                        :   {{AuthenticatorResponse/clientDataJSON}}
+                        ::  A new {{ArrayBuffer}}, created using |global|'s [=%ArrayBuffer%=], containing the bytes of
+                            <code>|credentialCreationData|.[=clientDataJSONResult=]</code>.
+
+                        :   {{AuthenticatorAttestationResponse/attestationObject}}
+                        ::  |attestationObject|
+
+                    :   {{PublicKeyCredential/clientExtensionResults}}
+                    ::  A new {{AuthenticationExtensions}} object associated with |global| containing
+                        <code>|credentialCreationData|.[=extensionOutputsMap=]</code>'s value.
+
+                        Note: <code>|credentialCreationData|.[=extensionOutputsMap=]</code> is an [=ordered map=] whose
+                            [=map/keys=] are all of type [=extension identifiers=] and [=map/values=] are all of type
+                            [=client extension output=]. Thus the latter is implicitly a [=record type=], which is the
+                            {{AuthenticationExtensions}}'s type.
+
+                1.  Return |pubKeyCred|.
+
+            1. [=set/For each=] remaining |authenticator| in |issuedRequests| invoke the [=authenticatorCancel=] operation on
+                |authenticator| and [=set/remove=] it from |issuedRequests|.
+
+            1. Return |constructCredentialAlg| and terminate this algorithm.
+
+            Issue: Do we need to replicitly return both |constructCredentialAlg| and |credentialCreationData| here?
```

Comment:
I believe that it's sufficient to return only *constructCredentialAlg*, on the theory that algorithm variables are captured by any subroutines that use them and only garbage-collected when all capturing routines are done. The main thing you have to guarantee is that no variable is used and modified in parallel, which we have to do by inspection whether or not you explicitly return the set of captured variables.

Reply:
ok, thanks, I'll remove that "Issue:"